File: VRTInterstitialCustomEventAppLovin.m
#import <AppLovinSDK/AppLovinSDK.h>
#import "VRTInterstitialCustomEventAppLovin.h"

//AppLovin Interstitial Adapter, Vrtcal as Primary
@interface VRTInterstitialCustomEventAppLovin() <MAAdDelegate>
@property MAInterstitialAd *maIntersatitialAd;
@end


@implementation VRTInterstitialCustomEventAppLovin

- (void) loadInterstitialAd {
    NSString *maAdUnitId = [self.customEventConfig.thirdPartyCustomEventData objectForKey:@"adUnitId"];
    self.maIntersatitialAd = [[MAInterstitialAd alloc] initWithAdUnitIdentifier:maAdUnitId];
    self.maIntersatitialAd.delegate = self;
    [self.maIntersatitialAd loadAd];
}

- (void) showInterstitialAd {
    [self.maIntersatitialAd showAd];
}

#pragma mark - MAAdDelegate
- (void)didLoadAd:(MAAd *)ad {
    [self.customEventLoadDelegate customEventLoaded];
}


- (void)didClickAd:(nonnull MAAd *)ad {
    [self.customEventShowDelegate customEventClicked];
}

- (void)didDisplayAd:(nonnull MAAd *)ad {
    [self.customEventShowDelegate customEventDidPresentModal:VRTModalTypeInterstitial];
    [self.customEventShowDelegate customEventShown];
}

- (void)didFailToDisplayAd:(nonnull MAAd *)ad withError:(nonnull MAError *)error {
    // No Analog
}

- (void)didFailToLoadAdForAdUnitIdentifier:(nonnull NSString *)adUnitIdentifier withError:(nonnull MAError *)error {
    VRTError *vrtError = [VRTError errorWithCode:VRTErrorCodeCustomEvent message:error.message];
    [self.customEventLoadDelegate customEventFailedToLoadWithError:vrtError];
}

- (void)didHideAd:(nonnull MAAd *)ad {
    [self.customEventShowDelegate customEventDidDismissModal:VRTModalTypeUnknown];
}



@end

File: VRTInterstitialCustomEventAppLovin.swift

import AppLovinSDK
import VrtcalSDK

//AppLovin Interstitial Adapter, Vrtcal as Primary

class VRTInterstitialCustomEventAppLovin: VRTAbstractInterstitialCustomEvent {
    private var maIntersatitialAd: MAInterstitialAd?
    private var maadDelegatePassthrough = MAAdDelegatePassthrough()
    
    override func loadInterstitialAd() {
        VRTLogInfo()
        
        
        let dict = customEventConfig.thirdPartyCustomEventData
        guard let maAdUnitId = dict["adUnitId"] as? String else {
            customEventLoadDelegate?.customEventFailedToLoad(
                vrtError: VRTError(
                    vrtErrorCode: .customEvent,
                    message: "Unable to extract adUnitId: \(dict)"
                )
            )
            return
        }
        
        maIntersatitialAd = MAInterstitialAd(adUnitIdentifier: maAdUnitId)
        
        maadDelegatePassthrough.customEventLoadDelegate = customEventLoadDelegate
        maadDelegatePassthrough.customEventShowDelegate = customEventShowDelegate
        
        maIntersatitialAd?.delegate = maadDelegatePassthrough
        maIntersatitialAd?.load()
    }

    override func showInterstitialAd() {
        VRTLogInfo()
        maIntersatitialAd?.show()
    }
}
